pallets/pips/src/lib.rs

```diff
@@ -1325,8 +1346,15 @@ impl<T: Config> Module<T> {
     ) -> ProposalState {
         <Proposals<T>>::mutate(id, |proposal| {
             if let Some(ref mut proposal) = proposal {
-                if (proposal.state, new_state) != (ProposalState::Pending, ProposalState::Scheduled)
-                {
+                // Decrement active count,
+                // unless transition is `(Pending | Scheduled) -> Scheduled`.
+                if !matches!(
+                    (proposal.state, new_state),
+                    (
+                        ProposalState::Pending | ProposalState::Scheduled,
+                        ProposalState::Scheduled,
+                    ),
+                ) {
                     Self::decrement_count_if_active(proposal.state);
                 }
```

Comment:
`decrement_count_if_active` is already checking if the current state `proposal.state` is active.

```suggestion
                // Decrement active count, if the `new_state` is not active.
                if !Self::is_active(new_state) {
                    Self::decrement_count_if_active(proposal.state);
                }
```

Reply:
Nice 👍